share/completions/npm.fish

```diff
@@ -106,6 +123,14 @@ for c in 'up' 'update'
     complete -f -c npm -n "__fish_npm_using_command $c" -s g -l global -d 'Update global package(s)'
 end
 
+# run
+command npm run | command tail -n +2 | command sed "s/^ *//" | while read -l name
+  set -l trim 20
+  read -l value
+  complete -f -c npm -n "__fish_npm_using_command run" -a (printf "%s " "$name") -d (echo "$value" | cut -c1-$trim)
+  complete -f -c npm -n "__fish_npm_using_command run-script" -a (printf "%s " "$name") -d (echo "$value" | cut -c1-$trim)
+end
```

Comment:
Oh, wait a minute... I don't think that works like you think.

That `command npm run | ...` stuff will be executed _once_, when the completion is auto-loaded!

Now, if I understand correctly, `npm run` is dependent on the current directory, so it'll trim and complete only the first directory you call the completion in (for a given shell instance).